pandas/core/frame.py

```diff
@@ -5272,10 +5272,15 @@ def combine(self, other, func, fill_value=None, overwrite=True):
         2  NaN  3.0 1.0
         """
         other_idxlen = len(other.index)  # save for compare
+        other_columns = other.columns
 
         this, other = self.align(other, copy=False)
         new_index = this.index
 
+        if other.empty and self.empty:
```

Comment:
Is this correct for dataframes that are empty becuase they have no columns?

On master we have

```python
In [11]: pd.DataFrame(index=['a', 'b']).combine_first(pd.DataFrame(index=['b', 'c']))
Out[11]:
Empty DataFrame
Columns: []
Index: [a, b, c]
```

What do you have on your branch? Shouldn't this be reindexing both the index and columns?